DQM/EcalMonitorDbModule/interface/EcalDQMStatusWriter.h

```diff
@@ -11,14 +11,14 @@
 
 #include <fstream>
 
-class EcalDQMStatusWriter : public edm::EDAnalyzer {
+class EcalDQMStatusWriter : public edm::one::EDAnalyzer<> {
 public:
   EcalDQMStatusWriter(edm::ParameterSet const &);
   ~EcalDQMStatusWriter() override {}
 
 private:
   void analyze(edm::Event const &, edm::EventSetup const &) override;
-  void beginRun(edm::Run const &, edm::EventSetup const &) override;
+  void beginRun(edm::Run const &, edm::EventSetup const &);
```

Comment:
Out of curiosity: why not `override` any more?

Reply:
Once I made the change from `EDAnalyzer` to `one::EDAnalyzer` for the legacy module warning, on compilation it gave the error that it does not override.

Reply:
Indeed,,,
@makortel @Dr15Jones do you have any idea why? There are several examples with a virtual `beginRun` overridden in one::EDAnalyzer in cmssw

Reply:
Because `one::EDAnalyzer<>` does not have a `beginRun` method. See 

https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkOneModuleInterface

Reply:
Thank you @Dr15Jones , that makes sense
What I don't understand is why there are examples with it, e.g. HLTrigger/HLTanalyzers/plugins/HLTrigReport.h, and apparently they compile...

Reply:
Although `edm::one::EDAnalyzer<>` does not have a `beginRun`, `edm::one::EDAnalyzer<edm::RunCache<..>>` does have a `beginRun` method.

Reply:
This also means that framework will not call this `beginRun()`, and given the code in
https://github.com/cms-sw/cmssw/blob/efb932182c1c2e47c0b78fdf9793ab08f8c68364/DQM/EcalMonitorDbModule/plugins/EcalDQMStatusWriter.cc#L24-L31
there is a high chance that the module doesn't work properly now.

Reply:
@abhih1 @jfernan2 in the PR description it is written "validated by running the relval workflow 136.874 using the runTheMatrix script": given the comment above from @makortel, can anyone confirm that this is really the case, and that the output did correspond to what expected?

Reply:
Should `EcalDQMStatusWriter` be run in workflow 136.874? It was not part of the "EDModules ran in runTheMatrix" that I extracted in spring.

Reply:
Hi I did validate it with the workflow 136.874 because that's the usual one we use for Ecal modules but now not sure if it is called in that. Do you happen to know where it might be called? If it is not ran in any of the runTheMatrix workflows I'm afraid I'm not sure how to validate this.

Reply:
The only place where this module seems to be loaded is in `DQM/​EcalCommon/​data/​writeEcalDQMStatus.py`. But that looks as if it needs to run at P5. Perhaps @depasse has a suggestion how to test the code.

Reply:
How about adding `edm::one::WatchRuns` as template argument to have the `beginRun()` function called?
`class EcalDQMStatusWriter : public edm::one::EDAnalyzer<edm::one::WatchRuns> {`

Reply:
Hi @tanmaymudholkar Do you happen to know where this file is used and if we can test it? I have checked in the P5 codes, but doesnt seem like we run it in any workflow.

Reply:
> Hi @tanmaymudholkar Do you happen to know where this file is used and if we can test it? I have checked in the P5 codes, but doesnt seem like we run it in any workflow.

Hi @abhih1 : I believe this module is used to update the record of known problematic towers on DB; these values are read in by ECAL DQM and used to conditionally flag towers or crystals as anomalies on DQM. I don't think they should be part of any workflow.